File: 020922/Code/USACO_21Dec_Bronze_1_demo.py
n = int(input())
s = input()

def count(ls):
    g_cnt = 0
    for i in ls:
        if i == 'G':
            g_cnt += 1
    if g_cnt == 1 or len(ls) - g_cnt == 1:
        return True
    else:
        return False

# length = 3, 4, 5 .. n

res = 0
for l in range(3, n + 1):
    # starts from i
    # n = 8
    # 0 1 2 3 4 5 6 7
    # l = 3
    # i -> [0, 5] == [0, 6)
    for i in range(n - l + 1):
        if count(s[i:i+l]):
            res += 1

print(res)

File: 020922/Code/USACO_21Dec_Bronze_1_demo.cpp
//http://www.usaco.org/index.php?page=viewproblem2&cpid=1155

#include<bits/stdc++.h>
using namespace std;

int n;

char s[505050];

bool count(int st, int ed){
	int g_cnt = 0;
	for (int i = st; i < ed; i++){
		if (s[i] == 'G') g_cnt++;
	}
	if (g_cnt == 1 || ed - st - g_cnt == 1){
		return true;
	}
	else{
		return false;
	}
}

int main(){
	scanf("%d", &n);
	scanf("%s", s);
	
	int res = 0;
	
	for (int l = 3; l <= n; l++){
		for (int i = 0; i <= n - l; i++){
			if (count(i, i + l)){
				res++;
			}
		}
	}
	
	printf("%d\n", res);
	
	return 0;
}
